tools/advisory-parser/src/models.rs

```diff
@@ -0,0 +1,182 @@
+//! Advisory data models and deserialization logic.
+//! This module defines the structure of security advisory TOML files and implements
+//! custom deserialization with validation for CVE and GHSA identifiers.
+
+use regex::Regex;
+use serde::de::{self, MapAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::fmt;
+
+#[derive(Deserialize)]
+pub struct Advisory {
+    pub advisory: AdvisoryInfo,
+}
+
+pub struct AdvisoryInfo {
+    #[allow(dead_code)]
+    /// Unique identifier for the Bottlerocket Security Advisory
+    pub id: String,
+
+    #[allow(dead_code)]
+    /// Title describing the security issue
+    pub title: String,
+
+    #[allow(dead_code)]
+    /// CVE identifier (e.g., "CVE-2024-1234")
+    pub cve: Option<String>,
+
+    #[allow(dead_code)]
+    /// GitHub Security Advisory identifier (e.g., "GHSA-xxxx-xxxx-xxxx")
+    pub ghsa: Option<String>,
+
+    #[allow(dead_code)]
+    /// Severity level (e.g., "high", "medium", "low")
+    pub severity: String,
+
+    #[allow(dead_code)]
+    /// Short description of the security issue
+    pub description: String,
+
+    /// List of affected packages with their patched versions
+    pub products: Vec<Product>,
+}
+
+impl<'de> Deserialize<'de> for AdvisoryInfo {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct AdvisoryInfoVisitor;
+
+        impl<'de> Visitor<'de> for AdvisoryInfoVisitor {
+            type Value = AdvisoryInfo;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("struct AdvisoryInfo")
+            }
+
+            fn visit_map<M>(self, mut map: M) -> std::result::Result<Self::Value, M::Error>
+            where
+                M: MapAccess<'de>,
+            {
+                let mut id = None;
+                let mut title = None;
+                let mut cve = None;
+                let mut ghsa = None;
+                let mut severity = None;
+                let mut description = None;
+                let mut products = None;
+
+                while let Some(key) = map.next_key::<String>()? {
+                    match key.as_str() {
+                        "id" => id = Some(map.next_value()?),
+                        "title" => title = Some(map.next_value()?),
+                        "cve" => {
+                            let value: String = map.next_value()?;
+                            // Reject non-ASCII characters (e.g., en dash U+2013, em dash U+2014)
+                            if !value.is_ascii() {
+                                return Err(de::Error::custom("CVE contains non-ASCII characters"));
+                            }
+                            // CVE format: CVE-YYYY-NNNNN where YYYY is year, NNNNN is 4+ digits
+                            let re = Regex::new(r#"^CVE-\d{4}-(0\d{3}|[1-9]\d{3,})$"#).unwrap();
```

Comment:
Can you make this regex pattern a static variable outside of the scope here - so that it can only be compiled once. 

You can use `lazy_static` for it.

Reply:
Can you also double check on the regex pattern. I found https://github.com/CVEProject/cve-schema/blob/main/schema/CVE_Record_Format.json#L55 from the official schema.

So maybe 
```suggestion
                            let re = Regex::new(r#"^CVE-[0-9]{4}-[0-9]{4,19}$"#).unwrap();
```

Reply:
Another thing is `unwrap` is discouraged as it causes panic with no proper error message. Can you use `unwrap_or` or `expect()` or other relevant function to properly handle the error when regex pattern fails to compile.

Reply:
added lazy static on the top and took care of these comments together.